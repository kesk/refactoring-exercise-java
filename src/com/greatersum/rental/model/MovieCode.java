package com.greatersum.rental.model;

public enum MovieCode {
    NEW,
    REGULAR,
    CHILDRENS
}
