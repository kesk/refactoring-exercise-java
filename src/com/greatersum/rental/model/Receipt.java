package com.greatersum.rental.model;

import java.math.BigDecimal;
import java.util.ArrayList;

public class Receipt {
    private Customer customer;
    private int frequentRenterPoints;
    private BigDecimal amount;
    private ArrayList<ReceiptItem> rentedMovies;

    public Receipt(Customer customer) {
        this.customer = customer;
        this.frequentRenterPoints = 0;
        this.amount = BigDecimal.valueOf(0);
        this.rentedMovies = new ArrayList<>();
    }

    public void incrementFrequentRenterPoints() {
        frequentRenterPoints++;
    }

    public void addReceiptItem(ReceiptItem item) {
        rentedMovies.add(item);
    }

    @Override
    public String toString() {
        BigDecimal totalAmount = BigDecimal.valueOf(0);
        String result = "Rental Record for " + customer.getName() + "\n";

        for (ReceiptItem ri : rentedMovies) {
            result += "\t" + ri.getMovie().getTitle() + "\t" + ri.getAmount() + "\n";
            totalAmount = totalAmount.add(ri.getAmount());
        }
        result += "Amount owed is " + totalAmount + "\n";
        result += "You earned " + frequentRenterPoints + " frequent renter points\n";
        return result;
    }
}
