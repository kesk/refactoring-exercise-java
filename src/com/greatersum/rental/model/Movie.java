package com.greatersum.rental.model;

import com.greatersum.rental.model.MovieCode;

public class Movie {
    private String id;
    private String title;
    private MovieCode code;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public MovieCode getCode() {
        return code;
    }

    public void setCode(MovieCode code) {
        this.code = code;
    }
}
