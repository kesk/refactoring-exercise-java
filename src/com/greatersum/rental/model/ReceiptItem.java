package com.greatersum.rental.model;

import com.greatersum.rental.model.Movie;

import java.math.BigDecimal;

public class ReceiptItem {
    private Movie movie;
    private BigDecimal amount;

    public ReceiptItem(Movie movie, BigDecimal amount) {
        this.movie = movie;
        this.amount = amount;
    }

    public Movie getMovie() {
        return movie;
    }

    public BigDecimal getAmount() {
        return amount;
    }
}
