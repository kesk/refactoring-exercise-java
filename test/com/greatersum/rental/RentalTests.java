package com.greatersum.rental;

import com.greatersum.rental.model.Customer;
import com.greatersum.rental.model.MovieRental;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.List;

public class RentalTests {
    @Test
    public void MartinTest() throws FileNotFoundException {
        List<MovieRental> rentals = Arrays.asList(
                new MovieRental("F001", 3),
                new MovieRental("F002", 1)
        );
        Customer customer = new Customer("martin", rentals);

        RentalInfo info = new RentalInfo();
        String actualResult = info.statement(customer);

        Approvals.verify(actualResult);
    }

    @Test
    public void AllMovieCodesFourDays() throws FileNotFoundException {
        List<MovieRental> rentals = Arrays.asList(
                new MovieRental("F001", 4),
                new MovieRental("F002", 4),
                new MovieRental("F003", 4),
                new MovieRental("F004", 4)
        );
        Customer customer = new Customer("martin", rentals);

        RentalInfo info = new RentalInfo();
        String actualResult = info.statement(customer);

        Approvals.verify(actualResult);
    }

    @Test
    public void AllMovieCodesOneDay() throws FileNotFoundException {
        List<MovieRental> rentals = Arrays.asList(
                new MovieRental("F001", 1),
                new MovieRental("F002", 1),
                new MovieRental("F003", 1),
                new MovieRental("F004", 1)
        );
        Customer customer = new Customer("martin", rentals);

        RentalInfo info = new RentalInfo();
        String actualResult = info.statement(customer);

        Approvals.verify(actualResult);
    }
}
