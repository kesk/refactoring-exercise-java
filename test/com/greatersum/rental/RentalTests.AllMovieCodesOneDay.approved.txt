Rental Record for martin
	Ran	2
	Trois Couleurs: Bleu	2
	Cars 2	1.5
	Latest Hit Release	3
Amount owed is 8.5
You earned 4 frequent renter points
